pkg/virt-controller/services/template.go

```diff
@@ -1244,7 +1244,7 @@ func generatePodAnnotations(vmi *v1.VirtualMachineInstance) (map[string]string,
 
 	annotationsSet[podcmd.DefaultContainerAnnotationName] = "compute"
 
-	multusAnnotation, err := GenerateMultusCNIAnnotation(vmi)
+	multusAnnotation, err := GenerateMultusCNIAnnotation(vmi, networkNameScheme)
```

Comment:
I find it unreasonable to pass this information through all this endless stack.
Until this PR, there was no dependency on such external data.

Now that there is such a dependency, all functions in the call stack had to change their signature, which is both noisy and not scalable (i.e. with another dependency, we will repeat this again).
It can also be examined with the known need to remove the legacy handling in the future, making this noise repeat.

I can think of two options to solve this:
1. Simple: Assume new scheme, if old naming is detected, override the multus annotation **after** the "regular" pod spec generation.
2. Advanced: Refactor & objectize the pod spec generation such that it is instantiated with arbitrary data and then each step can use the data per its needs.

I think we can work on option 1 at this stage.